parts/motor.go

```diff
@@ -0,0 +1,40 @@
+package parts
+
+import "fmt"
+
+type Motor struct {
+	On  bool
+	Rpm int
+}
+
+func (m *Motor) SetOn(on bool) {
+	m.On = on
+}
+
+func (m *Motor) IsOn() bool {
+	return m.On
+}
+
+func (m *Motor) SetRPM(rpm int, sb SeatbeltManipulator) error {
+	if rpm < 0 {
+		return fmt.Errorf("Negative RPM")
+	}
+	if rpm > 0 && sb.GetState() == SeatbeltStateEngauged {
+		m.Rpm = rpm
+	}
+	return fmt.Errorf("Seatbelt must be engauged to increase RPM")
+}
+
+func (m *Motor) GetRPM() int {
+	return m.Rpm
+}
+
+func (m *Motor) SetWheelsRPM(wm WheelsManipulator) error {
```

Comment:
I am not a big fan of the motor having a direct method for setting wheel speed.
I would think that most of these functions would be made at the car level.
Though this certainly does work.